packages/kbn-securitysolution-io-ts-alerting-types/src/actions/index.ts

```diff
@@ -29,6 +29,39 @@ export const RuleActionUuid = NonEmptyString;
 export type RuleActionParams = t.TypeOf<typeof RuleActionParams>;
 export const RuleActionParams = saved_object_attributes;
 
+export const RuleActionAlertsFilter = t.strict({
```

Comment:
@ersin-erdal thoughts on this?

Reply:
Actually they are not mix of `null` and `undefined`, they are just nullable.
https://github.com/elastic/kibana/blob/68a85504e1e46ef122c367cb7a402b9bd207ddbf/x-pack/plugins/alerting/common/rule.ts#L89

And `{ query: null, timeframe: null }` is not possible, there is a validation in the rules_client for that case, one of them has to be populated.
https://github.com/elastic/kibana/blob/68a85504e1e46ef122c367cb7a402b9bd207ddbf/x-pack/plugins/alerting/server/rules_client/lib/validate_actions.ts#L144

I followed a pattern here we generally use for the objects that have multiple child objects and at least one of them is mandatory, IIRC it was to prevent the object to be broken on partial update. But i think they could've been optional as well in this case.

As they are nullable in the create rule route's schema:
https://github.com/elastic/kibana/blob/68a85504e1e46ef122c367cb7a402b9bd207ddbf/x-pack/plugins/alerting/server/routes/lib/actions_schema.ts#L32
sending them from the UI was OK, they would be set as null automatically, but I forgot that security solutions has its own routes :) 

You can change that interface in this PR and make them optional if you want, or create a follow on issue so i can pick it up and change.

Reply:
@ersin-erdal thank you for the explanation. I was curious if there is some reason behind or hidden meaning and the question was mostly derived from TS interface definitions like [this one](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/common/rule.ts#L173). Presence of `null` and `undefined` in JS and as the side effect in TS makes it confusing because mostly people tend to consider them as value absence while sometimes `null` has some special meaning. Taking into account it's not always well documented it can be confusing. Due to that uncertainty some projects prohibit usage of `null` by a linter rule to make the codebase straightforward. There are obvious reasons why we can't do this in Kibana right now but it looks like we could simplify the codebase eventually.